pandas/core/base.py

```diff
@@ -470,9 +470,12 @@ def is_any_frame() -> bool:
             try:
                 result = DataFrame(result)
             except ValueError:
```

Comment:
@jreback I'm not super certain on what you meant by "dispatch differently", but in this particular case the `ValueError` is raised when the argument of `df.agg` is dictionary with scalar-value:

Fore example `df.agg({'name':'count'})` raises the ValueError but `df.agg({'name':['count']})` does not